Tests/BuildTests/PluginsBuildPlanTests.swift

```diff
@@ -79,4 +79,23 @@ final class PluginsBuildPlanTests: XCTestCase {
             )
         }
     }
+    
+    func testDocCPluginForBinaryDependency() async throws {
+        #if !os(macOS)
+        // binaryTarget/xcframework is only supported on Darwin platform
+        throw XCTSkip("Test requires macOS")
+        #endif
+        
+        try await fixture(name: "Miscellaneous/Plugins/SymbolGraphForBinaryDependency") { fixturePath in
+            let result = try await AsyncProcess.popen(arguments: [
+                fixturePath.appending(RelativePath("FooKit/Scripts/archive_xcframework.sh")).pathString,
+                "FooKit"
+            ])
+            try print(result.utf8Output())
+            try print(result.utf8stderrOutput())
+            XCTAssertEqual(result.exitStatus, .terminated(code: 0))
+            // Before we add -F support for xcframework, this call will throw since the command will abort with a non-zero exit code
+            let _ = try await executeSwiftPackage(fixturePath, extraArgs: ["generate-symbol-graph"])
```

Comment:
question: can we add an assertion that validates the expected behaviour of this step? Maybe `XCTAssertNoThrow` or check the return code?

Reply:
I'd like to to. But there is a question at present about if we should keep xcframework or not.

I originally thought it is required by the reviewer suggestion. And this is some blocking issues. See comments here https://github.com/swiftlang/swift-package-manager/pull/7998#issuecomment-2399356924

And the latest suggestion is we do not need them. And I'll probably delete this in next commit. See https://github.com/swiftlang/swift-package-manager/pull/7998#issuecomment-2443198351

Reply:
> issue (blocking): Add assertion to express test intent.

There is a exitStatus check to verify it. - `XCTAssertEqual(result.exitStatus, .terminated(code: 0))`

If the build is failing, the status code would abort with a non-zero exit code.